perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -2156,6 +2204,15 @@ public void deleteAttribute(PerunSession sess, AttributeDefinition attribute) th
 		//Remove services' required attributes
 		//TODO
 
+		//Remove attribute dependencies
+		synchronized (dependenciesMonitor) {
+			dependencies.remove(attribute);
+			strongDependencies.remove(attribute);
+			inverseDependencies.remove(attribute);
+			inverseStrongDependencies.remove(attribute);
+			allDependencies.remove(attribute);
+		}
```

Comment:
You should also remove all occurrences of this "attribute" from the values (not just from the keys).

Reply:
Done.